### DISCUSSION TITLE
Nested objects/arrays in requests and responses

### DISCUSSION THREAD

**Discussion by ianberdin** [scalar/scalar#1835] [Ideas]
Hi,

I am extremely grateful for the beautiful, high-quality tool you have created for generating and managing documentation. I am absolutely thrilled.

I wanted to ask if you plan to support nested structures for input parameters and responses?

( look at [array][object][] )
<img width="604" alt="image" src="https://github.com/scalar/scalar/assets/2521933/b2dde4a9-b1d7-4c8b-9696-02e01ea7bf6f">

This example from Redoc, shows how a response can show nested types, including oneOf.
<img width="664" alt="image" src="https://github.com/scalar/scalar/assets/2521933/b27bd69a-c574-4888-afb2-58a3ca411ccd">

**Comment by marclave** [comment]
hi @ianberdin ! thanks for making this discussion!

would you be able to share part of or the openapi spec for this : ) we agree we can do a better job on nested objects/arrays in requests/responses : )
